go/worker/registration/worker.go

```diff
@@ -585,6 +618,81 @@ Loop:
 	}
 }
 
+func (w *Worker) metricsWorker() {
+	w.logger.Info("delaying metrics worker start until initial registration")
+	select {
+	case <-w.stopCh:
+		return
+	case <-w.ctx.Done():
+		return
+	case <-w.initialRegCh:
+	}
+
+	w.logger.Debug("starting metrics worker")
+
+	t := time.NewTicker(periodicMetricsInterval)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-w.stopCh:
+			return
+		case <-w.ctx.Done():
+			return
+		case <-t.C:
+		}
+
+		// Update metrics.
+		epoch, err := w.beacon.GetEpoch(w.ctx, consensus.HeightLatest)
+		if err != nil {
+			w.logger.Warn("unable to query epoch", "err", err)
+			continue
+		}
+		status, err := w.GetRegistrationStatus(w.ctx)
+		if err != nil {
+			w.logger.Warn("unable to get registration status", "err", err)
+			continue
+		}
+		nodeStatus := status.NodeStatus
+		if nodeStatus == nil {
+			w.logger.Debug("skipping node status metrics, empty node status")
+			continue
+		}
+
+		// Frozen metric.
+		switch nodeStatus.IsFrozen() {
+		case true:
+			workerNodeStatusFrozen.Set(1)
+		case false:
+			workerNodeStatusFrozen.Set(0)
+		}
+
+		// Election eligible metric.
+		switch {
+		case nodeStatus.ElectionEligibleAfter == 0:
+			workerNodeRegistrationEligible.Set(0)
+		case nodeStatus.ElectionEligibleAfter >= epoch:
+			workerNodeRegistrationEligible.Set(0)
+		default:
+			workerNodeRegistrationEligible.Set(1)
+		}
+
+		// Runtime metrics.
+		for id, faults := range nodeStatus.Faults {
+			rtId := id.String()
+			workerNodeStatusFaults.WithLabelValues(rtId).Set(float64(faults.Failures))
```

Comment:
This will never reset when the counter goes back to zero and the entry in the `Faults` field is removed for that runtime. It should probably set these metrics for any configured runtimes that have no corresponding faults entry.

Reply:
Yeah makes sense. Updated